logic.js

```diff
@@ -0,0 +1,215 @@
+let posts = [
+  {
+    text: 'First post!',
+    id: 'p1',
+    comments: [
+      { id: 'c1', text: 'First comment on first post!' },
+      { id: 'c2', text: 'Second comment on first post!!' },
+      { id: 'c3', text: 'Third comment on first post!!!' }
+    ]
+  },
+  {
+    text: 'Aw man, I wanted to be first',
+    id: 'p2',
+    comments: [
+      { id: 'c4', text: "Don't wory second poster, you'll be first one day." },
+      { id: 'c5', text: 'Yeah, believe in yourself!' },
+      { id: 'c6', text: 'Haha second place what a joke.' }
+    ]
+  }
+]
+
+const newLocal = 'tweetweData'
+if (!localStorage.getItem(newLocal)) {
+  saveData()
+} else {
+  posts = JSON.parse(localStorage.getItem(newLocal))
+}
+
+
+function saveData () {
+  localStorage.setItem(newLocal, JSON.stringify(posts))
+}
+
+let postIdCounter = posts.length + 1
+
+let commentIdCounter = 0
+for (let i = 0; i < posts.length; i++) {
+  commentIdCounter += posts[i].comments.length
+}
+
+function getPosts () {
+  return posts
+}
+
+function addPost (text) {
+  const newPost = {}
+  newPost.text = text
+  newPost.id = `p${postIdCounter}`
+  newPost.comments = []
+  posts.push(newPost)
+  postIdCounter++
+  saveData()
+  return posts
+}
+
+function removePost (postID) {
+  for (let i = 0; i < posts.length; i++) {
+    if (posts[i].id === postID) {
+      posts.splice(i, 1)
+    }
+  }
+  saveData()
+  return posts
+}
+// addPost("Hello")
+// addPost("My own post")
+// addPost('nika pika')
+
+// removePost('p1')
+// console.log(posts)
+
+function addComment (text, postID) {
+  const newComment = {}
+  commentIdCounter++
+  newComment.id = 'c' + commentIdCounter
+  newComment.text = text
+  for (let i = 0; i < posts.length; i++) {
+    if (posts[i].id === postID) {
+      posts[i].comments.push(newComment)
+    }
+  }
+  saveData()
+  return posts
+}
+
+function removeComment (postID, commentID) {
+  for (let i = 0; i < posts.length; i++) {
+    if (posts[i].id === postID) {
+      for (let j = 0; j < posts[i].comments.length; j++) {
+        if (posts[i].comments[j].id === commentID) {
+          posts[i].comments.splice(j, 1)
+        }
+      }
+    }
+  }
+  saveData()
+  return posts
+}
+
+// eslint-disable-next-line no-unused-vars
+function Tweeter () {
+  return {
+    addPost,
+    removePost,
+    addComment,
+    removeComment,
+    getPosts
+  }
```

Comment:
When creating a module, all of the methods should be written inside the module, in the code above there is no reason for using the module, because the functions are global functions, not incapsulate inside the module